go/chat/partsource.go

```diff
@@ -0,0 +1,108 @@
+package chat
```

Comment:
sorry to nit but can this be named `participantsSource.go`? I find `part` confusing, care less about the variable names in this file since it's internal but the explicit filename is more clear to me